### PR TITLE
#863g4r03x: Add plugin for newsletter subscription via keycloak

### PR DESCRIPTION
**Pull request by atennert**
* moved blacklist into separate blacklist package
* added comments and documentation
* set version to v0.0.5

### CODE REVIEW

File: src/main/java/com/qupaya/sha1/Sha1User.java [hunk 1]
```diff
@@ -1,5 +1,10 @@
 package com.qupaya.sha1;
 
+/*
+ * Apparently whatever framework Keycloak uses, doesn't like POJOs
+ * for automatic conversion from JSON data when they are written
+ * in Kotlin. So, here it is in Java.
```
Comment (Markus-Ende) [inline src/main/java/com/qupaya/sha1/Sha1User.java:6]: 🫠

File: src/main/kotlin/com/qupaya/brevo/NewsletterRegistrationEventListenerProvider.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+        /*
+         * Yes, we could use the Brevo API to create a (DOI) contact, but
+         * using the form data POST request ensures that handling like
+         * success pages can be controlled inside Brevo.
+         */
+        val http = HttpClient.newHttpClient()
+        val request = HttpRequest.newBuilder()
+            .uri(URI.create(brevoFormLink))
+            .POST(HttpRequest.BodyPublishers.ofString(getFormDataAsString(formData)))
```
Comment (Markus-Ende) [inline src/main/kotlin/com/qupaya/brevo/NewsletterRegistrationEventListenerProvider.kt:72]: don't we need error handling here? Should we maybe add a retry?

**Review by Markus-Ende — APPROVED** [review]
LGTM, just a comment

File: src/main/kotlin/com/qupaya/brevo/NewsletterRegistrationEventListenerProvider.kt (hunk 2 above)
Comment (atennert) [reply src/main/kotlin/com/qupaya/brevo/NewsletterRegistrationEventListenerProvider.kt:72]: done :)

**Pull request merged**